src/chain_sync/sync_status.rs

```diff
@@ -0,0 +1,245 @@
+    /// The epoch of the heaviest fully validated tipset on the node's main chain.
+    /// This shows overall node progress, distinct from fork-specific progress.
+    pub(crate) validated_chain_head_epoch: ChainEpoch,
+    /// When processing for this fork started.
+    pub(crate) start_time: Option<DateTime<Utc>>,
+    /// Last time status for this fork was updated.
+    pub(crate) last_updated: Option<DateTime<Utc>>,
+}
+
+/// Contains information about the current status of the node's synchronization process.
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq, JsonSchema)]
+pub struct SyncStatusReport {
+    /// Overall status of the node's synchronization.
+    status: NodeSyncStatus,
+    /// The epoch of the heaviest validated tipset on the node's main chain.
+    current_head_epoch: ChainEpoch,
+    /// The tipset key of the current heaviest validated tipset.
+    #[schemars(with = "crate::lotus_json::LotusJson<TipsetKey>")]
+    #[serde(with = "crate::lotus_json")]
+    current_head_key: Option<TipsetKey>,
+    // Current highest epoch on the network.
+    network_head_epoch: ChainEpoch,
+    /// Estimated number of epochs the node is behind the network head.
+    /// Can be negative if the node is slightly ahead, due to estimation variance.
+    epochs_behind: i64,
+    /// List of active fork synchronization tasks the node is currently handling.
+    active_forks: Vec<ForkSyncInfo>,
+    /// When the node process started.
+    node_start_time: DateTime<Utc>,
+    /// Last time this status report was generated.
+    last_updated: DateTime<Utc>,
+}
+
+lotus_json_with_self!(SyncStatusReport);
+
+impl SyncStatusReport {
+    pub(crate) fn init() -> Self {
+        Self {
+            node_start_time: Utc::now(),
+            ..Default::default()
+        }
+    }
+
+    pub(crate) fn set_current_chain_head_key(&mut self, tipset_key: TipsetKey) {
+        self.current_head_key = Some(tipset_key);
+    }
+
+    pub(crate) fn set_current_chain_head_epoch(&mut self, epoch: ChainEpoch) {
+        self.current_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_current_chain_head_epoch(&self) -> ChainEpoch {
+        self.current_head_epoch
+    }
+
+    pub(crate) fn set_network_head(&mut self, epoch: ChainEpoch) {
+        self.network_head_epoch = epoch;
+    }
+
+    pub(crate) fn get_network_head_epoch(&self) -> ChainEpoch {
+        self.network_head_epoch
+    }
+
+    pub(crate) fn get_current_chain_head_key(&self) -> Option<&TipsetKey> {
+        self.current_head_key.as_ref()
+    }
+
+    pub(crate) fn set_epochs_behind(&mut self, epochs_behind: i64) {
+        self.epochs_behind = epochs_behind;
+    }
+
+    pub(crate) fn get_epochs_behind(&self) -> i64 {
+        self.epochs_behind
+    }
+
+    pub(crate) fn set_status(&mut self, status: NodeSyncStatus) {
+        self.status = status;
+    }
+
+    pub(crate) fn get_status(&self) -> NodeSyncStatus {
+        self.status.clone()
```

Comment:
clone can be avoided here

Reply:
I think this is negligible since `NodeSyncStatus` is an enum which is represented as integers in memory, anyways added a copy trait in the enum and removed the `clone`.